### ISSUE TITLE
IntelliSense: void main() { constexpr const char* strs[] = { "" }; } gives "expression must have a constant value"

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#2939]
In a .cpp file use code
```
void main() { constexpr const char* strs[] = { "" }; }
```
This compiles with cl, gcc, and clang, but IntelliSense reports "expression must have a constant value: attempt to access run-time storage".

It repros with VS as well: https://developercommunity.visualstudio.com/content/problem/410833/c-intellisense-void-main-constexpr-const-char-strs.html .

**Comment by AndrewGaspar** [comment]
I'm running into this with using `__func__` and `__PRETTY_FUNCTION__` in gcc and clang in constexpr contexts.

In my case, this is in the context of a macro where the values _could_ be non-constexpr, and I'd be okay with "lying" to VS Code to get the red squiggles to go away. Is there any way to detect Intellisense in the C Preprocessor?

**Comment by AndrewGaspar** [comment]
For example, I'd like to write something like
```c++
#ifdef INTELLISENSE
#define MY_CONSTEXPR
#else
#define MY_CONSTEXPR constexpr
#endif
```

**Comment by AndrewGaspar** [comment]
Ah, heh, I was pretty close: https://stackoverflow.com/questions/26724244/is-there-a-macro-defined-when-intellisense-is-parsing

**Comment by bialpio** [comment]
I'm running into the same (?) issue when using VS Code with IntelliSense on Chrome's code base.

The red squiggles appear for example under `TRACE_EVENT_BEGIN0`, `TRACE_EVENT_END0` & `TRACE_EVENT0` (see for example https://cs.chromium.org/chromium/src/third_party/blink/renderer/modules/vr/vr_display.cc?type=cs&q=TRACE_EVENT_BEGIN0&sq=package:chromium&g=0&l=771).

The code compiles fine with clang. I'm using compile_commands.json to configure IntelliSense, `intelliSenseMode` is set to `clang-x64`, with `--sysroot` added to `compilerPath` to work around issue #1575.

**Comment by sean-mcmanus** [comment]
This bug was fixed with 0.26.0. Let us know if anyone is still hitting it.

Our next release will have some other fixes involving an incorrect "expression must have a constant value" in other scenarios (see https://developercommunity.visualstudio.com/content/problem/280487/constexpr-constructor-with-initializer-list.html ).
